ovos_utils/file_utils.py

```diff
@@ -0,0 +1,240 @@
+    found = resolve_ovos_resource_file(res_name)
+    if found:
+        return found
+
+    # Finally look for it in the source package
+    paths = [
+        "/opt/venvs/mycroft-core/lib/python3.7/site-packages/",  # mark1/2
+        "/opt/venvs/mycroft-core/lib/python3.4/site-packages/ ",
+        # old mark1 installs
+        "/home/pi/mycroft-core"  # picroft
+    ]
+    if root_path:
+        paths += [root_path]
+    for p in paths:
+        filename = os.path.join(p, 'mycroft', 'res', res_name)
+        filename = os.path.abspath(os.path.normpath(filename))
+        if os.path.isfile(filename):
+            return filename
+
+    return None  # Resource cannot be resolved
+
+
+def read_vocab_file(path):
+    """ Read voc file.
+
+        This reads a .voc file, stripping out empty lines comments and expand
+        parentheses. It returns each line as a list of all expanded
+        alternatives.
+
+        Args:
+            path (str): path to vocab file.
+
+        Returns:
+            List of Lists of strings.
+    """
+    vocab = []
+    with open(path, 'r', encoding='utf8') as voc_file:
+        for line in voc_file.readlines():
+            if line.startswith('#') or line.strip() == '':
+                continue
+            vocab.append(expand_options(line.lower()))
+    return vocab
+
+
+def load_regex_from_file(path, skill_id):
+    """Load regex from file
+    The regex is sent to the intent handler using the message bus
+
+    Args:
+        path:       path to vocabulary file (*.voc)
+        skill_id:   skill_id to the regex is tied to
+    """
+    from ovos_utils.intents.intent_service_interface import munge_regex
+
+    regexes = []
+    if path.endswith('.rx'):
+        with open(path, 'r', encoding='utf8') as reg_file:
+            for line in reg_file.readlines():
+                if line.startswith("#"):
+                    continue
+                LOG.debug('regex pre-munge: ' + line.strip())
+                regex = munge_regex(line.strip(), skill_id)
+                LOG.debug('regex post-munge: ' + regex)
+                # Raise error if regex can't be compiled
+                try:
+                    re.compile(regex)
+                    regexes.append(regex)
+                except Exception as e:
+                    LOG.warning(f'Failed to compile regex {regex}: {e}')
+
+    return regexes
+
+
+def load_vocabulary(basedir, skill_id):
+    """Load vocabulary from all files in the specified directory.
+
+    Args:
+        basedir (str): path of directory to load from (will recurse)
+        skill_id: skill the data belongs to
+    Returns:
+        dict with intent_type as keys and list of list of lists as value.
+    """
+    from ovos_utils.intents.intent_service_interface import to_alnum
+
+    vocabs = {}
+    for path, _, files in walk(basedir):
+        for f in files:
+            if f.endswith(".voc"):
+                vocab_type = to_alnum(skill_id) + splitext(f)[0]
```

Comment:
`dismissal_skill.chancencounternevermind`?

Reply:
its the same munging mycroft does,just following their api